test/node-api.js

```diff
@@ -6,11 +6,188 @@ chai.use(codeEquality);
+
+    expect(output).to.eql([
+      {
+        type: "template-tag",
+        tagName: "template",
+        contents: "Hello!",
+        range: {
+          start: 12,
+          end: 39
+        },
+        contentRange: {
+          start: 22,
+          end: 28
+        },
+        startRange: {
+          start: 12,
+          end: 22
+        },
+        endRange: {
+          start: 28,
+          end: 39
+        }
+      },
+    ]);
+  });
+
+  it("inside class body", function() {
+    let p = new Preprocessor();
+    let output = p.parse(`
+      class A {
+        <template>Hello!</template>
+      }
+    `);
+
+    expect(output).to.eql([
+      {
+        type: "template-tag",
+        tagName: "template",
+        contents: "Hello!",
+        range: {
+          start: 25,
+          end: 52
+        },
+        contentRange: {
+          start: 35,
+          end: 41
+        },
+        startRange: {
+          start: 25,
+          end: 35
+        },
+        endRange: {
+          start: 41,
+          end: 52
+        }
+      },
+    ]);
+  });
+
+  it("preceded by a slash character", function() {
+    let p = new Preprocessor();
+    // What is this testing?
+    // Would a better test be:
+    // `const divide = 1 / <template>Hello!</template>;`
+    let output = p.parse(`
+      const divide = () => 4 / 2;
+      <template>Hello!</template>
+    `);
+
+    expect(output).to.eql([
+      {
+        type: "template-tag",
+        tagName: "template",
+        contents: "Hello!",
+        range: {
+          start: 41,
+          end: 68
+        },
+        contentRange: {
+          start: 51,
+          end: 57
+        },
+        startRange: {
+          start: 41,
+          end: 51
+        },
+        endRange: {
+          start: 57,
+          end: 68
+        }
+      },
+    ]);
+  });
+
+  it("/<template>/ inside a regexp", function() {
+    let p = new Preprocessor();
+    let output = p.parse(`
+      const myregex = /<template>/;
+      <template>Hello!</template>
+    `);
+
+    expect(output).to.eql([
+      {
+        type: "template-tag",
+        tagName: "template",
+        contents: "Hello!",
+        range: {
+          start: 43,
+          end: 70
+        },
+        contentRange: {
+          start: 53,
+          end: 59
+        },
+        startRange: {
+          start: 43,
+          end: 53
+        },
+        endRange: {
+          start: 59,
+          end: 70
+        }
+      },
+    ]);
+  });
+
+  it("no match", function() {
+    let p = new Preprocessor();
+    let output = p.parse("console.log('Hello world');");
+
+    expect(output).to.eql([]);
+  });
+
+  it("Emits parse errors", function() {
+    let p = new Preprocessor();
+    expect(function() {
+      p.process(
+        `const thing = "face";
+  <template>Hi`,
+        "path/to/my/component.gjs"
+      );
+    }).to.throw(`Parse Error at path/to/my/component.gjs:2:15: 2:15`);
+  });
+});
```

Comment:
do you think it's worth adding a test for a multi-line `<template>` tag? 

and then maybe one with:

```js
<template>
  y
  <template>x</template>
  z
</template>
```

This should parse out to 1 set of ranges

Reply:
That is actually not legal. The first `</template>` closes the content tag. Because content-tag is deliberately agnostic to the interior language, it doesn't know anything about delimiter pairing.